### PR TITLE
add: product detail

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
xem chi tiết sản phẩm 
- https://edu-redmine.sun-asterisk.vn/issues/74318

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/service/product.service.ts [hunk 1]
```diff
@@ -1,20 +1,62 @@
...
+
+  public async getProductDetail(id: number) {
+    try {
+      const product = await this.productRepository.findOne({
+        where: { id },
+        loadRelationIds: {
+          relations: ['type'],
+        },
+      });
+      if (!product) {
```
Comment (huongnt-2545) [inline src/service/product.service.ts]: method `findOne` return null nếu ko có bản ghi thỏa mãn nên chỗ này ko cần check null nữa e hè

File: src/service/product.service.ts [hunk 2]
```diff
@@ -1,20 +1,62 @@
...
+
+  public async getToppings() {
+    try {
+      const products = await this.productRepository.find({
+        where: { type: Equal(TYPE_PRODUCT.TOPPING) },
+        loadRelationIds: {
+          relations: ['type'],
+        },
+      });
+      if (!products) {
```
Comment (huongnt-2545) [inline src/service/product.service.ts]:
```suggestion
      if (products?.length) {
```

File: src/views/layouts/main.hbs [hunk 3]
```diff
@@ -46,4 +47,4 @@
   <script src='/js/home/home.js'></script>
 </body>
 
-</html>
+</html>
```
Comment (huongnt-2545) [inline src/views/layouts/main.hbs]: end file

File: src/views/product/index.hbs [hunk 4]
```diff
@@ -39,4 +41,4 @@
     </div>
 
 </div>
-<script src='/js/product/product.js'></script>
+<script src='/js/product/product.js'></script>
```
Comment (huongnt-2545) [inline src/views/product/index.hbs]: end file

File: src/views/product/product-detail.hbs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                        </div>
+                    </div>
+                </div>
+                {{/each}}
+            </div>
+        </div>
+    </div>
+</div>
+
+<script src='/js/product/product-detail.js'></script>
```
Comment (huongnt-2545) [inline src/views/product/product-detail.hbs]: end file

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**